third_party/build_deps.sh

```diff
@@ -0,0 +1,283 @@
+#!/bin/bash -xe
+
+ROOT_DIR=$(pwd)
+export INSTALL_PREFIX=${INSTALL_PREFIX:-/usr/local}
+export CC_COMP=${CC_COMP:-gcc}
+export CXX_COMP=${CXX_COMP:-g++}
+echo ${INSTALL_PREFIX}
+echo ${CC_COMP}
+echo ${CXX_COMP}
+echo ${CMAKE_TARGET_ARCH}
+echo ${BUILD_ARCH_OPTION}
+echo ${HOST_ARCH_OPTION}
+echo ${SYSROOT_ARG}
+echo ${WITH_FFMPEG}
+echo ${EXTRA_PROTOBUF_FLAGS}
+echo ${OPENCV_TOOLCHAIN_FILE}
+echo ${EXTRA_FLAC_FLAGS}
+echo ${EXTRA_LIBSND_FLAGS}
+
+#zlib
+pushd third_party/zlib
+CFLAGS="-fPIC" \
+CXXFLAGS="-fPIC" \
+CC=${CC_COMP} \
+CXX=${CXX_COMP} \
+./configure --prefix=${INSTALL_PREFIX}
+make -j"$(grep ^processor /proc/cpuinfo | wc -l)"
+make install
+popd
+
+# CMake
+pushd third_party/CMake
+./bootstrap --parallel=$(grep ^processor /proc/cpuinfo | wc -l)
+make -j"$(grep ^processor /proc/cpuinfo | wc -l)"
+make install
+popd
+
+# protobuf, make two steps for cross compilation
+pushd third_party/protobuf
+./autogen.sh
+./configure CXXFLAGS="-fPIC" --prefix=/usr/local --disable-shared 2>&1 > /dev/null
+make -j"$(grep ^processor /proc/cpuinfo | wc -l)" 2>&1 > /dev/null
+make install 2>&1 > /dev/null
+# only when cross compiling
+if [ "${CC_COMP}" != "gcc" ]; then
+  make clean
+      ./autogen.sh
+      ./configure \
+      CXXFLAGS="-fPIC ${EXTRA_PROTOBUF_FLAGS}" \
+      CC=${CC_COMP} \
+      CXX=${CXX_COMP} \
+      ${HOST_ARCH_OPTION} \
+      ${BUILD_ARCH_OPTION} \
+      ${SYSROOT_ARG} \
+      --with-protoc=/usr/local/bin/protoc \
+      --prefix=${INSTALL_PREFIX}
+  make -j$(nproc) install
+fi
+popd
+
+# LMDB
+pushd third_party/lmdb/libraries/liblmdb/
+patch -p3 < ${ROOT_DIR}/patches/Makefile-lmdb.patch
+  CFLAGS="-fPIC" CXXFLAGS="-fPIC" CC=${CC_COMP} CXX=${CXX_COMP} prefix=${INSTALL_PREFIX} \
+make -j"$(grep ^processor /proc/cpuinfo | wc -l)"
+make install
+popd
+
+# libjpeg-turbo
+pushd third_party/libjpeg-turbo/
+echo "set(CMAKE_SYSTEM_NAME Linux)" > toolchain.cmake
+echo "set(CMAKE_SYSTEM_PROCESSOR ${CMAKE_TARGET_ARCH})" >> toolchain.cmake
+echo "set(CMAKE_C_COMPILER ${CC_COMP})" >> toolchain.cmake
+  CFLAGS="-fPIC" \
+  CXXFLAGS="-fPIC" \
+  CC=${CC_COMP} \
+  CXX=${CXX_COMP} \
```

Comment:
```suggestion
  CFLAGS="-fPIC" \
  CXXFLAGS="-fPIC" \
  CC=${CC_COMP} \
  CXX=${CXX_COMP} \
```
```suggestion
CFLAGS="-fPIC" \
CXXFLAGS="-fPIC" \
CC=${CC_COMP} \
CXX=${CXX_COMP} \
```
nitpick: weird indent

Reply:
The is will be indistinctable from other lines, the point here is to show that it a part of the following command.